python/dgl/nn/pytorch/network_emb.py

```diff
@@ -0,0 +1,220 @@
+                    wid += 1
+        else:
+            for nodetype in self.g.ntypes:
+                self.type2id[nodetype] = dict()
+                for idx in range(self.g.num_nodes(nodetype)):
+                    self.type2id[nodetype][idx] = wid
+                    self.id2word[wid] = self.nid2word[nodetype][idx]
+                    wid += 1
+
+        ## convert the edge metapath to the node metapath
+        edge2des = {}
+        edge2src = {}
+        for comb in self.g.canonical_etypes:
+            edge2des[comb[1]] = comb[2]
+            edge2src[comb[1]] = comb[0]
+        nodespath = []
+        nodespath.append(edge2src[self.metapath[0]])
+        for edge in self.metapath:
+            nodespath.append(edge2des[edge])
+
+        ##start random walk
+        for idx in tqdm.trange(self.g.number_of_nodes(nodespath[0])):
+            traces = random_walk(g=self.g, nodes=[idx, ] * self.node_repeat, metapath=self.metapath)
+            for tr in traces[0].cpu().numpy():
+                line = [self.type2id[nodespath[i]][tr[i]] for i in range(len(tr))]
+                self.walk_dataset.append(line)
+                if len(line) > 1:
+                    for word in line:
+                        self.token_count += 1
+                        word_frequency[word] = word_frequency.get(word, 0) + 1
+
+        # Filter out the number of words less than the minimum frequency
+        t = 0.0001
+        for word, freq in word_frequency.items():
+            if freq > self.min_count:
+                word_frequency_filtered[word] = freq
+                f = np.array(freq) / self.token_count
+                self.discards[word] = np.sqrt(t / f) + (t / f)
+
+        # get the number of all nodes
+        self.word_count = self.g.number_of_nodes()
+        self.word_frequency = word_frequency_filtered
+
+        print("Total embeddings: " + str(self.word_count))
+        print("real embeddings：" + str(len(self.word_frequency)))
+
+        self.u_embeddings = nn.Embedding(self.word_count, self.emb_dimension, sparse=self.sparse)
+        self.v_embeddings = nn.Embedding(self.word_count, self.emb_dimension, sparse=self.sparse)
+
+    def _initTableNegatives(self):
+        # get a table for negative sampling, if word with index 2 appears twice, then 2
+        # will be listed in the table twice.
+        pow_frequency = np.array(list(self.word_frequency.values())) ** 0.75
+        words_pow = sum(pow_frequency)
+        ratio = pow_frequency / words_pow
+        count = np.round(ratio * self.NEGATIVE_TABLE_SIZE)
+        word_list = list(self.word_frequency.keys())
+        for wid, c in enumerate(count):
+            self.negatives += [word_list[wid]] * int(c)
+        self.negatives = np.array(self.negatives)
+        np.random.shuffle(self.negatives)
+        self.sampling_prob = ratio
+
+    def initParameters(self):
+        """
+        Initialize the embedding parameters
+        """
+        initrange = 1.0 / self.emb_dimension
```

Comment:
```suggestion
        init_range = 1.0 / self.emb_dimension
```

update the use below accordingly